sig-architecture/production-readiness.md

```diff
@@ -41,6 +41,62 @@ aspects of the process.
   with and without the feature are necessary. At the very least, think about
   conversion tests if API types are being modified.
 
+
+#### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+* **What signals should trigger rollback?**
+  Focus on metrics and treat other signals (like logs) as last resort.
+
+* **Were upgrade and rollback tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+
+#### Monitoring requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+* **How can an operator determine if the feature is functioning properly?**
+  Focus on metrics that cluster operators may gather from different
+  components and treat other signals as last resort.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  - [ ] Metrics
+    - Metric name:
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+  - [ ] Other (treat as last resort)
+    - Details:
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
```

Comment:
Provide some examples.  This can be very difficult for people to describe and it varies widely for different types of features.  server-side-apply, versus protobuf encoding, versus CRIO enablement, versus user-space proxy, versus priority and fairness, versus cronjob controller.  They all have differently shaped objectives.

Reply:
This is not a new question - I just moved the section around for more logical flow.
For now I added a TODO to not block this PR - will send out a separate PR later this week.

Reply:
Added something. I think we can iterate further on it, but I think it's already better than it was.